core/src/main/java/com/sequenceiq/cloudbreak/controller/validation/template/TemplateValidator.java

```diff
@@ -157,13 +159,24 @@ private void validateVolumeType(VolumeTemplate value, Platform platform, Validat
         }
     }
 
+    private boolean isIDBrokerInstanceGroup(InstanceGroup instanceGroup) {
+        return GROUP_NAME_ID_BROKER.equalsIgnoreCase(instanceGroup.getGroupName());
+    }
+
     private void validateVolumeCount(VolumeTemplate value, VmType vmType, VolumeParameterType volumeParameterType,
-        ValidationResult.ValidationResultBuilder validationBuilder) {
+        ValidationResult.ValidationResultBuilder validationBuilder, InstanceGroup instanceGroup) {
         if (vmType != null && needToCheckVolume(volumeParameterType, value.getVolumeCount())) {
             VolumeParameterConfig config = vmType.getVolumeParameterbyVolumeParameterType(volumeParameterType);
             if (config != null) {
                 if (value.getVolumeCount() > config.maximumNumber()) {
                     validationBuilder.error(String.format("Max allowed volume count for '%s': %s", vmType.value(), config.maximumNumber()));
+                } else if (isIDBrokerInstanceGroup(instanceGroup)) {
+                    if (value.getVolumeCount() != 0)  {
+                        // IDBroker does not use data volume, so its volume count should be zero
+                        validationBuilder.error(
+                                String.format("IDBroker volume count: %s is not zero. It does't use data volume, and its count must be zero.",
```

Comment:
You have a typo in `doesn't`

I prefer not using contractions in error logs. Use the full form `does not`